pkg/reconciler/v1alpha1/broker/broker.go

```diff
@@ -80,94 +88,99 @@ type ReconcilerArgs struct {
-			err = c.Watch(&source.Kind{Type: t}, &handler.EnqueueRequestForOwner{OwnerType: &v1alpha1.Broker{}, IsController: true})
-			if err != nil {
-				return nil, err
-			}
-		}
+	serviceInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: controller.Filter(v1alpha1.SchemeGroupVersion.WithKind("Broker")),
+		Handler:    reconciler.Handler(impl.EnqueueControllerOf),
+	})
 
-		return c, nil
-	}
-}
+	deploymentInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: controller.Filter(v1alpha1.SchemeGroupVersion.WithKind("Broker")),
+		Handler:    reconciler.Handler(impl.EnqueueControllerOf),
+	})
 
-// InjectClient implements controller runtime's inject.Client.
-func (r *reconciler) InjectClient(c client.Client) error {
-	r.client = c
-	return nil
+	return impl
 }
 
 // Reconcile compares the actual state with the desired, and attempts to
 // converge the two. It then updates the Status block of the Broker resource
 // with the current status of the resource.
-func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	ctx := context.TODO()
-	ctx = logging.WithLogger(ctx, r.logger.With(zap.Any("request", request)))
-
-	broker := &v1alpha1.Broker{}
-	err := r.client.Get(ctx, request.NamespacedName, broker)
-
-	if k8serrors.IsNotFound(err) {
-		logging.FromContext(ctx).Info("Could not find Broker")
-		return reconcile.Result{}, nil
+func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
+	// Convert the namespace/name string into a distinct namespace and name
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		r.Logger.Errorf("invalid resource key: %s", key)
+		return nil
 	}
 
-	if err != nil {
-		logging.FromContext(ctx).Error("Could not Get Broker", zap.Error(err))
-		return reconcile.Result{}, err
+	// Get the Broker resource with this namespace/name
+	original, err := r.brokerLister.Brokers(namespace).Get(name)
+	if apierrs.IsNotFound(err) {
+		// The resource may no longer exist, in which case we stop processing.
+		logging.FromContext(ctx).Error("broker key in work queue no longer exists", zap.Any("key", key))
+		return nil
+	} else if err != nil {
+		return err
 	}
 
-	originalReadiness := broker.Status.IsReady()
+	// Don't modify the informers copy
+	broker := original.DeepCopy()
 
-	// Reconcile this copy of the Broker and then write back any status updates regardless of
-	// whether the reconcile error out.
-	reconcileErr := r.reconcile(ctx, broker)
-	if reconcileErr != nil {
-		logging.FromContext(ctx).Error("Error reconciling Broker", zap.Error(reconcileErr))
-		r.recorder.Event(broker, corev1.EventTypeWarning, brokerReconcileError, fmt.Sprintf("Broker reconcile error: %v", reconcileErr))
+	// Reconcile this copy of the Broker and then write back any status
+	// updates regardless of whether the reconcile error out.
+	err = r.reconcile(ctx, broker)
```

Comment:
Leave this as `reconcileErr`, it doesn't match the standard go `err` semantics. Namely that it is important needs to be retained after the next if statement.

Reply:
Hmm, all other controllers have it as err. Do you want me to change it anyway?

Reply:
Yes! As-is, changing `:=` to `=` on line 173 will cause the program to continue to compile and work correctly most of the time, but will fail occasionally.

Reply:
Done!

Reply:
Actually, I left err there, and change the update status to updateStatusErr...

Reply:
I originally did that too, but switched to `reconcileErr`, because `reconcileErr` is the non-normal behavior of error, while `updateStatusErr` is the more standard, check immediately and then ignore.

Reply:
OK, makes sense. Will do in another PR. We should do the same across all controllers to maintain consistency.

Reply:
Totally agree. If you want, I'm happy to make that clean up PR.

Reply:
OK, go for it so that I can focus on having the registry PR ready. Thanks Adam!